services/api/src/resources/task/task_definition_resolvers.ts

```diff
@@ -249,6 +272,16 @@ export const addAdvancedTaskDefinition = async (
     project
   );
 
+  if (input.deployTokenInjection == undefined) {
```

Comment:
So for these I think you could do something like this
`input.deployTokenInjection = input.deployTokenInjection || false;`